### PR TITLE
5.0.0 template for vpc block csi driver - compute identity and snapshot

### PR DESCRIPTION
**Pull request by GunaKKIBM**

### CODE REVIEW

File: config-templates/ibm/ibm-vpc-block-csi-driver/5.0/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,723 @@
+apiVersion: v1
+kind: List
+metadata:
+  name: ibm-vpc-block-csi-driver
+  namespace: kube-system
+  annotations:
+    version: 5.0
+    revision: 1
```
Comment (prankulmahajan) [inline config-templates/ibm/ibm-vpc-block-csi-driver/5.0/deployment.yaml:8]: @mssachan Please confirm if revision is required for new templates released

File: config-templates/ibm/ibm-vpc-block-csi-driver/5.0/metadata.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
+{
+  "name": "5.0",
+  "revision": "1",
```
Comment (prankulmahajan) [inline config-templates/ibm/ibm-vpc-block-csi-driver/5.0/metadata.json:3]: > @mssachan Please confirm if revision is required for new templates released

@mssachan Here also

**Review by prankulmahajan — COMMENTED** [review]
Looks good.

File: config-templates/ibm/ibm-vpc-block-csi-driver/5.0/metadata.json [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      "value": "For support contact IBM"
+    },
+    {
+      "name": "supported-envs",
+      "value": "IBM"
+    },
+    {
+      "name": "supported-k8s-versions",
+      "value": "1.22,1.23,1.24,1.25"
+    }
```
Comment (GunaKKIBM) [inline config-templates/ibm/ibm-vpc-block-csi-driver/5.0/metadata.json:47]: Add supported ocp versions

File: config-templates/ibm/ibm-vpc-block-csi-driver/5.0/metadata.json [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    {
+      "name": "support-msg",
+      "value": "For support contact IBM"
+    },
+    {
+      "name": "supported-envs",
+      "value": "IBM"
+    },
+    {
+      "name": "supported-k8s-versions",
```
Comment (GunaKKIBM) [inline config-templates/ibm/ibm-vpc-block-csi-driver/5.0/metadata.json]: remove k8s versions

File: config-templates/ibm/ibm-vpc-block-csi-driver/5.0/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,732 @@
...
+    spec:
+      serviceName: "ibm-vpc-block-service"
+      replicas: 1
+      selector:
+        matchLabels:
+          app: ibm-vpc-block-csi-driver
+      template:
+        metadata:
+          labels:
+            app: ibm-vpc-block-csi-driver
```
Comment (GunaKKIBM) [inline config-templates/ibm/ibm-vpc-block-csi-driver/5.0/deployment.yaml:547]: Add label for version.

File: config-templates/ibm/ibm-vpc-block-csi-driver/5.0/deployment.yaml (hunk 1 above)
Comment (mssachan) [reply config-templates/ibm/ibm-vpc-block-csi-driver/5.0/deployment.yaml:8]: its required

File: config-templates/ibm/ibm-vpc-block-csi-driver/5.0/deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,734 @@
...
+              secret:
+                secretName: storage-secret-store
+  - kind: StatefulSet
+    apiVersion: apps/v1
+    metadata:
+      name: ibm-vpc-block-csi-controller
+      namespace: kube-system
+      labels:
+        app: ibm-vpc-block-csi-driver
+      annotations:
```
Comment (mssachan) [inline config-templates/ibm/ibm-vpc-block-csi-driver/5.0/deployment.yaml:536]: do we need to add this `version` under `annotations` field or it should come under `labels`?

File: config-templates/ibm/ibm-vpc-block-csi-driver/5.0/deployment.yaml [hunk 7]
```diff
@@ -0,0 +1,734 @@
...
+        name: ibm-vpc-block-controller-sa
+        namespace: kube-system
+    roleRef:
+      kind: ClusterRole
+      name: vpc-block-driver-snapshotter-role
+      apiGroup: rbac.authorization.k8s.io   
+  - kind: CSIDriver
+    apiVersion: storage.k8s.io/v1
+    metadata:
+      name: vpc.block.csi.ibm.io
```
Comment (mssachan) [inline config-templates/ibm/ibm-vpc-block-csi-driver/5.0/deployment.yaml:296]: Add label here also

File: config-templates/ibm/ibm-vpc-block-csi-driver/5.0/deployment.yaml [hunk 8]
```diff
@@ -0,0 +1,734 @@
...
+          iam_client_secret = "bx"
+          g2_token_exchange_endpoint_url = "{{{g2_token_exchange_endpoint_url}}}"
+          g2_riaas_endpoint_url = "{{{g2_riaas_endpoint_url}}}"
+          g2_resource_group_id = "{{{g2_resource_group_id}}}"
+          g2_api_key = "{{{g2_api_key}}}"
+          provider_type = "g2"
+    kind: Secret
+    metadata:
+      name: storage-secret-store
+      namespace: kube-system
```
Comment (mssachan) [inline config-templates/ibm/ibm-vpc-block-csi-driver/5.0/deployment.yaml:320]: Add label for this resource as well

**Review by mssachan — CHANGES_REQUESTED** [review]
PR looks good with minor comments

**Comment by mssachan** [comment]
Also please confirm which version should be set as default ? 4.3.0 or 5.0?

File: config-templates/ibm/ibm-vpc-block-csi-driver/5.0/deployment.yaml (hunk 6 above)
Comment (GunaKKIBM) [reply config-templates/ibm/ibm-vpc-block-csi-driver/5.0/deployment.yaml:536]: I checked with @ambiknai , she confirmed it should be under annotations

File: config-templates/ibm/ibm-vpc-block-csi-driver/5.0/deployment.yaml (hunk 8 above)
Comment (GunaKKIBM) [reply config-templates/ibm/ibm-vpc-block-csi-driver/5.0/deployment.yaml:320]: storage-secret-store is a resource used by other components too, Adding any label here wouldn't be correct is my opinion

File: config-templates/ibm/ibm-vpc-block-csi-driver/5.0/deployment.yaml (hunk 7 above)
Comment (GunaKKIBM) [reply config-templates/ibm/ibm-vpc-block-csi-driver/5.0/deployment.yaml:296]: Added

File: config-templates/ibm/ibm-vpc-block-csi-driver/5.0/deployment.yaml (hunk 8 above)
Comment (mssachan) [reply config-templates/ibm/ibm-vpc-block-csi-driver/5.0/deployment.yaml:320]: ok

**Review by mssachan — APPROVED** [review]
LGTM

**Pull request merged**